File: 18-frame-printer/frame_printer.php
<?php
// PHP program to print words in a frame
// Written by M.V.Harish Kumar on 18-09-2024

$s = readline("Enter a string: ");
$words = explode(" ", $s);

$max_len = 0;
foreach ($words as $word) {
	if (strlen($word) > $max_len)
		$max_len = strlen($word);
}

echo str_repeat('*', $max_len+4) . "\n";
foreach ($words as $word) {
	echo '* ' . str_pad($word, $max_len) . " *\n";
}
echo str_repeat('*', $max_len+4) . "\n";

File: 18-frame-printer/frame_printer.go
// Go program to print words in a frame
// Written by M.V.Harish Kumar on 18-06-2025
package main

import (
	"bufio"
	"fmt"
	"os"
	"slices"
	"strings"
)

func main() {
	scnr := bufio.NewScanner(os.Stdin)
	fmt.Print("Enter a string: ")
	scnr.Scan()
	words := strings.Fields(scnr.Text())
	maxLen := len(slices.MaxFunc(words, func(a, b string) int { 
		return len(a) - len(b)
	}))

	border := strings.Repeat("*", maxLen + 4)
	fmt.Println(border)
	for _, word := range words {
		fmt.Printf("* %-*s *\n", maxLen, word)
	}
	fmt.Println(border)
}
